src/api/Metadata.js

```diff
@@ -153,27 +153,220 @@ class Metadata extends File {
                 throw e;
             }
         }
-
-        return getProp(templates, 'data.entries', []).filter(
-            template => !template.hidden && template.templateKey !== METADATA_TEMPLATE_CLASSIFICATION,
-        );
+        return getProp(templates, 'data.entries', []);
     }
 
     /**
-     * Gets metadata instances for file
+     * Gets metadata instances for a box file
      *
      * @param {string} id - file id
      * @return {Object} array of metadata instances
      */
     async getInstances(id: string): Promise<Array<MetadataInstance>> {
-        return getProp(
-            await this.xhr.get({
+        this.errorCode = ERROR_CODE_FETCH_METADATA;
+        let instances = {};
+        try {
+            instances = await this.xhr.get({
                 url: this.getMetadataUrl(id),
                 id: getTypedFileId(id),
+            });
+        } catch (e) {
+            const { status } = e;
+            if (isUserCorrectableError(status)) {
+                throw e;
+            }
+        }
+        return getProp(instances, 'data.entries', []);
+    }
+
+    /**
+     * Returns a list of templates that can be added by the user.
+     * For collabed files, only custom properties is allowed.
+     *
+     * @return {Object} template for custom properties
+     */
+    getUserAddableTemplates(
+        customPropertiesTemplate: MetadataEditorTemplate,
+        enterpriseTemplates: Array<MetadataEditorTemplate>,
+        hasMetadataFeature: boolean,
+        isExternallyOwned?: boolean,
+    ): Array<MetadataEditorTemplate> {
+        let userAddableTemplates: Array<MetadataEditorTemplate> = [];
+        if (hasMetadataFeature) {
+            userAddableTemplates = isExternallyOwned
+                ? [customPropertiesTemplate]
+                : [customPropertiesTemplate].concat(enterpriseTemplates);
+        }
+        return userAddableTemplates.filter(template => template.templateKey !== METADATA_TEMPLATE_CLASSIFICATION);
+    }
+
+    /**
+     * Extracts classification for different representation in the UI.
+     *
+     * @param {string} id - box file id
+     * @param {Array} instances - metadata instances
+     * @return {Array} metadata instances without classification
+     */
+    extractClassification(id: string, instances: Array<MetadataInstance>): Array<MetadataInstance> {
+        const index = instances.findIndex(instance => instance.$template === METADATA_TEMPLATE_CLASSIFICATION);
```

Comment:
can you just use Array.prototype.find?

Reply:
I agree that's more readable to use `find` and then just make an `indexOf` call if you need it for the `splice` below.